orchagent/fdborch.cpp

```diff
@@ -34,21 +34,66 @@ void FdbOrch::update(sai_fdb_event_t type, const sai_fdb_entry_t* entry, sai_obj
 
         (void)m_entries.insert(update.entry);
         SWSS_LOG_DEBUG("FdbOrch notification: mac %s was inserted into vlan %d", update.entry.mac.to_string().c_str(), entry->vlan_id);
+        
+        for (auto observer: m_observers)
+        {
+            observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
+        }
+        
         break;
+        
     case SAI_FDB_EVENT_AGED:
-    case SAI_FDB_EVENT_FLUSHED:
     case SAI_FDB_EVENT_MOVE:
         update.add = false;
 
         (void)m_entries.erase(update.entry);
         SWSS_LOG_DEBUG("FdbOrch notification: mac %s was removed from vlan %d", update.entry.mac.to_string().c_str(), entry->vlan_id);
+        
+        for (auto observer: m_observers)
+        {
+            observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
+        }
+      
+        break;
+        
+    case SAI_FDB_EVENT_FLUSHED:
+        if( !bridge_port_id && !entry->vlan_id)
+        {
+            for(auto itr = m_entries.begin(); itr != m_entries.end(); ++itr )
+            {
+                /*This is a flush all case, need to clear up all the fdb entries*/
+                update.entry.mac = itr->mac;
+                update.entry.vlan = itr->vlan;
+                update.add = false;
+                
+                m_entries.erase(itr);
```

Comment:
Looks to me as a non-standard way of deleting the elements of a set in for-loop as it invalidates the iterator after erase. It might have worked in test. Can you check this?

Reply:
Will check. and I am revising the solution as Guohan suggested, will update all of these with the new solution.